### PR TITLE
Optional filter

### LINKED ISSUE #26: Option to turn off xkcd effect
[Issuehunt badges](https://issuehunt.io/r/timqian/chart.xkcd/issues/26)

For serious people
e.g. https://github.com/timqian/star-history/issues/59#issuecomment-526926972

Might also need some style updates after turning off

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $50.00)

- [issuehunt issuehunt](https://issuehunt.io/u/issuehunt) ($50.00)
### Submitted pull Requests
- [#28 Optional filter](https://issuehunt.io/r/timqian/chart.xkcd/pull/28)
---

### Tips

- Checkout the [Issuehunt explorer](https://issuehunt.io/r/timqian/chart.xkcd/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on IssueHunt to raise funds.
---
IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### PR DESCRIPTION
**Pull request by denisinvader**
**Related issue**

Fix #26

**Screenshot before and after this change**
- before
screenshot before
- after
screenshot after

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Referenced issues

This pull request has been submitted to:
- [#26: Option to turn off xkcd effect](https://issuehunt.io/repos/200573908/issues/26)
---

IssueHunt has been backed by the following sponsors. [Become a sponsor](https://issuehunt.io/membership/members)
</details>

### CODE REVIEW

File: docs/04-line.md [hunk 1]
```diff
@@ -44,3 +44,4 @@ const lineChart = new chartXkcd.Line(svg, {
     - up right: `chart.Xkcd.positionType.upLeft`
 - `dataColors`: array of colors for different datasets
 - `fontFamily`: customize font family used in the chart
+- `disableEffect`: disable xkcd effect for serious people (default `false`)
```
Comment (timqian) [inline docs/04-line.md]: About the option name, `unxkcdify` sounds better to me. What do you think?

File: src/Line.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ class Line {
       datasets,
     };
     this.options = options;
+    this.filter = !options.disableEffect ? 'url(#xkcdify)' : null;
     this.svgEl = select(svg)
       .style('stroke-width', '3')
       .style('font-family', this.options.fontFamily || 'xkcd')
```
Comment (timqian) [inline src/Line.js]: Besides turning off filter, turning off the 'xkcd' font seems necessary too

File: docs/04-line.md (hunk 1 above)
Comment (denisinvader) [reply docs/04-line.md]: I don't mind. If you wish so I will change the option name. 
Maybe `unXkcdify` to respect camelCase?

File: src/Line.js (hunk 2 above)
Comment (denisinvader) [reply src/Line.js]: I'm not sure. Regarding the original issue, there are issues with chart readability, but the font is rather readable in my opinion. Moreover, there is already the option to change font family

Comment (timqian) [reply src/Line.js]: Make sense, this is not mentioned in the original issue and not a must have feature. We can do this later if people want it

File: docs/04-line.md (hunk 1 above)
Comment (timqian) [reply docs/04-line.md]: `unxkcdify` should work, like `uncomment` it is one word

Comment (denisinvader) [reply docs/04-line.md]: Updated

**Pull request merged**